### PR TITLE
Document soft linking urn

### LINKED ISSUE #993: Document Linking in API specs through URN format
Teams have started migrating and want to use links in RAML that point to other types.  

But the ability to link through the URN format is entirely undocumented yet, not even a mention in the README or the smoke test sites.

### PR DESCRIPTION
**Pull request by davifantasia**
Closes #993 

https://github.com/commercetools/commercetools-docs-kit/issues/1001 to be fixed in a different PR

### CODE REVIEW

**Review by timonrey — APPROVED** [review]
lgtm

File: api-specs/test/types/links.raml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        description: A link pointing to a type in another microsite.
+        type: overrides.TestTypeOnAnotherMicrosite
+  LinkToTypeOnExternalSite:
+    properties:
+      LinkToTypeOnExternalSite:
+        description: A link pointing to a type in an external website.
+        type: overrides.TestTypeOnAnExternalWebsite
+  LinkFromDescription:
+    description: |-
+      Link [ArrayTestType](/types/general#ctp:test:type:ArrayTestType) with `[ArrayTestType](/types/general#ctp:test:type:ArrayTestType)`
```
Comment (nkuehn) [inline api-specs/test/types/links.raml:26]: This is not a link notation that we should use.   Use either URLs to headings or pure URNs.

File: api-specs/test/types/links.raml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    properties:
+      linkFromPropertyDescription:
+        description: |-
+          From property description, link [GET request for /{projectKey}/resource/key={key}](/endpoints/methods#ctp:test:endpoint:/%257BprojectKey%257D/resource/key=%257Bkey%257D:GET) with `[GET request for /{projectKey}/resource/key={key}](/endpoints/methods#ctp:test:endpoint:/%257BprojectKey%257D/resource/key=%257Bkey%257D:GET)`
```
Comment (nkuehn) [inline api-specs/test/types/links.raml:30]: Same here, this combination does not make sense

File: api-specs/test/types/links.raml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        type: string
+      linkWithMicrositeNotation:
+        description: |-
+          Link using microsite notation with `[Text](/../<microsite-name>/path#ctp:<api-name>:type:<type-name>)`
```
Comment (nkuehn) [inline api-specs/test/types/links.raml:34]: and again

File: packages/gatsby-theme-api-docs/README.md [hunk 4]
```diff
@@ -67,9 +67,23 @@ The file location determines the `apiKey` through which it can be addressed:
...
-If you want to override the location to link to for certain API data types, create a file called `type-locations.yaml` under the `src/data` folder of the website. This can be useful to choose manual markdown based documentation for certain types or to link ot shared types used on other websites like public standards. Use the following format to override the type locations:
+Soft links are autogenerated for a type that is used on an MDX page. This provides users ability to easily link from the field of one type, to the location where the type is rendered or defined.
+
+For types, the links are generated using our custom URN - `ctp:<api-key>:type:<type-name>`.
+
+The links can be used in markdown or description fields of types and resources like any other markdown link:
+
+- Using a full path: `[Text](/path/#ctp:<api-key>:type:<type-name>)`
+- Using microsite notation: `[Text]](/../<microsite-name>/path#ctp:<api-key>:type:<type-name>)`
+- Using absolute URL: `[Text](https://example.com/path#ctp:<api-key>:type:<type-name>)`
```
Comment (nkuehn) [inline packages/gatsby-theme-api-docs/README.md:80]: Also here.

File: packages/gatsby-theme-api-docs/README.md [hunk 5]
```diff
@@ -67,9 +67,23 @@ The file location determines the `apiKey` through which it can be addressed:
...
+
+For endpoints, the links are generated using our custom URN - `ctp:<api-key>:endpoint:<end-point>:<method>`. Note that `<end-point>` should includes URI encoded characters for `{`, `}`, `#`, `:`, etc.
```
Comment (nkuehn) [inline packages/gatsby-theme-api-docs/README.md:82]:
```suggestion
For endpoints, the links are generated using our custom URN - `ctp:<api-key>:endpoint:<endpoint>:<method>`. Note that `<endpoint>` should includes URI encoded characters for `{`, `}`, `#`, `:`, etc.
```

File: packages/gatsby-theme-api-docs/src/components/resource/method/method.js [hunk 6]
```diff
@@ -62,7 +62,7 @@ const Method = ({
       {title ? (
         <TitleWithAnchor id={id}>{title}</TitleWithAnchor>
       ) : (
-        <a name={id}></a>
+        <span id={id} />
```
Comment (nkuehn) [inline packages/gatsby-theme-api-docs/src/components/resource/method/method.js:65]: This causes a confusing merge conflict, but it's actually not a complex change

**Review by nkuehn — CHANGES_REQUESTED** [review]
The examples and test cases do not represent what the feature is intended for.  

The point of URN linking is not to have a worse way of hardwiring on what page a given endpoint or type is documented in.  The point of the feature is to be able to link to a type or endpoint without knowing on what page it is.

**Pull request closed without merging**